addons/guide/modifiers/guide_modifier_curve.gd

```diff
@@ -0,0 +1,55 @@
+@tool
+## Applies a separate curve to each input axis.
+class_name GUIDEModifierCurve
+extends GUIDEModifier
+
+
+## The curve to apply to the x axis
+@export var curve_x:Curve
```

Comment:
I'm not sure how often one needs to apply different curves to different parts of the input. I have no data on this but my gut feeling is that the most common use case is to apply a single curve on all or a subset of the input. Maybe this could be restructured like this:

```
@export var curve:Curve

@export var x:bool = true
@export var y:bool = true
@export var z:bool = true
```

This would apply the given curve to the parts of the input vector that are set to `true`.  This also has the benefit that if you only want to apply a single curve to all parts of the input, you don't have to make sure that all three curves are in sync. If you need to actually modify different input parts with different curves, you can still chain multiple modifiers. What do you think?